### PR TITLE
Replace strided slice with tf.expand_dims

### PR DESCRIPTION
**Pull request by jplu**
# What does this PR do?

This PR aims to replace the strided slice notation by its TF operator counterpart. As proposed by @mfuntowicz https://github.com/huggingface/transformers/pull/9890#discussion_r571939682

### CODE REVIEW

File: src/transformers/models/albert/modeling_tf_albert.py [hunk 1]
```diff
@@ -543,7 +543,7 @@ def call(
         # So we can broadcast to [batch_size, num_heads, from_seq_length, to_seq_length]
         # this attention mask is more simple than the triangular masking of causal attention
         # used in OpenAI GPT, we just need to prepare the broadcast dimension here.
-        extended_attention_mask = inputs["attention_mask"][:, tf.newaxis, tf.newaxis, :]
+        extended_attention_mask = tf.expand_dims(tf.expand_dims(inputs["attention_mask"], axis=1), axis=1)
```
Comment (sgugger) [inline src/transformers/models/albert/modeling_tf_albert.py:546]: Could we do a `tf.reshape` here? Two `expand_dims` in a row is a bit hard to read (and the result is not super understandable).

File: src/transformers/models/bert/modeling_tf_bert.py [hunk 2]
```diff
@@ -655,7 +655,7 @@ def call(
         # So we can broadcast to [batch_size, num_heads, from_seq_length, to_seq_length]
         # this attention mask is more simple than the triangular masking of causal attention
         # used in OpenAI GPT, we just need to prepare the broadcast dimension here.
-        extended_attention_mask = inputs["attention_mask"][:, tf.newaxis, tf.newaxis, :]
+        extended_attention_mask = tf.expand_dims(tf.expand_dims(inputs["attention_mask"], axis=1), axis=1)
```
Comment (sgugger) [inline src/transformers/models/bert/modeling_tf_bert.py:658]: Same here.

**Review by sgugger — APPROVED** [review]
Good with the change, thanks for doing it.
When two new dimensions are need, can we use a different operator to make the code more readable? Suggested `tf.reshape` but there might be something else available?
```
tf.expand_dims(tf.expand_dims(inputs["attention_mask"], axis=1), axis=1)
```
is not super readable.

**Comment by jplu** [comment]
> When two new dimensions are need, can we use a different operator to make the code more readable? Suggested tf.reshape but there might be something else available?

Yes, this is doable with `tf.reshape`.

**Review by patrickvonplaten — APPROVED** [review]

**Review by LysandreJik — APPROVED** [review]
LGTM, could you run the slow tests on the concerned models to make sure?

**Comment by jplu** [comment]
All the slow tests of the concerned models are ok!

**Pull request merged**